pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -351,7 +356,9 @@ func (cb *ClusterBuilder) buildCluster(name string, discoveryType cluster.Cluste
 			statPrefix := telemetry.BuildStatPrefix(cb.req.Push.Mesh.OutboundClusterStatName,
 				string(service.Hostname), subset, port, 0, &service.Attributes)
 
-			if statPrefix[len(statPrefix)-1:] != constants.ClusterAltStatNameDelimeter {
+			// Add the cluster name delimeter if it's not the last character.
+			isGE23 := pm.ParseIstioVersion(cb.proxyVersion).Minor >= 23
```

Comment:
we should probably just store the parsed version (`proxy.Version`) in the ClusterBuilder instead of re-parsing here.

We also used to have helpers like

```
// IsIstioVersionGE119 checks whether the given Istio version is greater than or equals 1.19.
func IsIstioVersionGE119(version *model.IstioVersion) bool {
	return version == nil ||
		version.Compare(&model.IstioVersion{Major: 1, Minor: 19, Patch: -1}) >= 0
}
```
that I tend to like, notably because it makes it easy to track where all of our temporary logic is